src/core/symbols/symbol.cpp

```diff
@@ -261,6 +264,26 @@ void Symbol::save(QXmlStreamWriter& xml, const Map& map) const
 	if (!description.isEmpty())
 		xml.writeTextElement(QLatin1String("description"), description);
 	saveImpl(xml, map);
+	if (!custom_icon.isNull())
+	{
+		QBuffer buffer;
+		QImageWriter writer{&buffer, QByteArrayLiteral("PNG")};
+		if (writer.write(custom_icon))
```

Comment:
AFAICT, `QImage::save(...)` will do. There are no special `QImageWiter` settings being applied.

Reply:
`QImage::save/load` offer no way to determine what kind of error occured in the process. 
(In the given function I could only throw an error at the moment. I chose to print a debug message instead.)